acme/acme.go

```diff
@@ -331,6 +334,14 @@ func (a *ACME) CreateLocalConfig(tlsConfig *tls.Config, checkOnDemandDomain func
 func (a *ACME) getCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
 	domain := types.CanonicalDomain(clientHello.ServerName)
 	account := a.store.Get().(*Account)
+	//use regex to test for wildcard certs that might have been added into TLSConfig
+	for k := range a.TLSConfig.NameToCertificate {
+		selector := "^" + strings.Replace(k, "*.", ".*\\.?", -1) + "$"
```

Comment:
Since you cannot have a double wildcard certificate, sometimes people issue certs with wildcard subdomains as SANS. Example: `*.mysite.com` could have a SAN of `*.foo.mysite.com`, in which case it is up to the certificate to cover the SANs. I also search the SANs of the certificate here as well, so if your cert has both `*.mysite.com` with a SAN of `*.foo.mysite.com`, it will be matched either way.